serialization.md

```diff
@@ -477,4 +477,76 @@ The resulting Tar archive for `f60c56784b83` has the following entries:
 /etc/.wh.my-app-config
 ```
 
+Whiteout files MUST only apply to resources in lower layers.
+Files that are present in the same layer as a whiteout file can only be hidden by whiteout files in subsequent layers.
+The following is a base layer with several resources:
+
+```
+a/
+a/b/
+a/b/c/
+a/b/c/bar
+```
+
+When the next layer is created, the original `a/b` directory is deleted and recreated with `a/b/c/foo`:
+
+```
+a/
+a/.wh..wh..opq
+a/b/
+a/b/c/
+a/b/c/foo
+```
+
+When processing the second layer, `a/.wh..wh..opq` is applied first, before creating the new version of `a/b, regardless of the ordering in which the whiteout file was encountered.
+For example, the following layer is equivalent to the layer above:
```

Comment:
This is correct. An opaque directory in one layer should only hide the lower layer and not apply to the currently layer. If the intent is just to remove the directory, an opaque directory is not needed and the directory can just be marked deleted. The usage of an opaque marker specifies that the intent is to hide the lower layer and allow for files in the current layer.